### PR TITLE
Est 110 possibilitar que coordenador/comissão possa consolidar inscrições

### PR DESCRIPTION
**Pull request by romulocosta100**

### CODE REVIEW

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java [hunk 1]
```diff
@@ -175,6 +178,49 @@ public String entrevista(@Valid @ModelAttribute("entrevista") Entrevista entrevi
...
+
+			Servidor servidor = servidorService.getServidorPorCpf(auth.getName());
+
+			List<Servidor> comissao = selecao.getMembrosComissao();
+
+			if(comissao.contains(servidor)){
+
+				List<Inscricao> inscricoes = inscricaoService.getInscricoesPorSelecao(idSelecao);
+				model.addAttribute("selecao", selecao);
+				model.addAttribute("inscricoes", inscricoes);
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Codacy Issue found: [The String "inscricoes" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2698980352/issues/source?bid=3244003&fileBranchId=3341467#l203)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java [hunk 2]
```diff
@@ -194,5 +193,34 @@ public void update(Integer idInscricao,boolean classificado) {
 		
 	}
 
+	@Override
+	@Transactional
+	public void consolidacaoDeTodos(Integer idSelecao, boolean consolidacao) {
+		Map<String, Object> params = new SimpleMap<String, Object>();
+		params.put("consolidacao", consolidacao);
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java]: Codacy Issue found: [The String "consolidacao" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2698980408/issues/source?bid=3244003&fileBranchId=3341467#l200)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java [hunk 3]
```diff
@@ -175,6 +178,49 @@ public String entrevista(@Valid @ModelAttribute("entrevista") Entrevista entrevi
...
+				model.addAttribute("todosConsolidados",inscricaoService.todosConsolidados(idSelecao));
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Codacy Issue found: [The String "todosConsolidados" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2698980352/issues/source?bid=3244003&fileBranchId=3341467#l204)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java [hunk 4]
```diff
@@ -175,6 +178,49 @@ public String entrevista(@Valid @ModelAttribute("entrevista") Entrevista entrevi
...
+		}else{
+
+			Servidor servidor = servidorService.getServidorPorCpf(auth.getName());
+
+			List<Servidor> comissao = selecao.getMembrosComissao();
+
+			if(comissao.contains(servidor)){
+
+				List<Inscricao> inscricoes = inscricaoService.getInscricoesPorSelecao(idSelecao);
+				model.addAttribute("selecao", selecao);
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Codacy Issue found: [The String "selecao" appears 5 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2698980352/issues/source?bid=3244003&fileBranchId=3341467#l202)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java [hunk 5]
```diff
@@ -175,6 +178,22 @@ public String entrevista(@Valid @ModelAttribute("entrevista") Entrevista entrevi
 		return REDIRECT_PAGINA_LISTAR_SELECAO;
 	}
 	
+	
+	@RequestMapping(value = "consolidarTodos", method = RequestMethod.GET,  produces="application/json")
+	public @ResponseBody String consolidarTodos(@RequestParam("idSelecao") Integer idSelecao,@RequestParam("consolidacao") boolean consolidacao){
+		inscricaoService.consolidacaoDeTodos(idSelecao, consolidacao);
+		return "{}";
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Codacy Issue found: [The String "{}" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2707229881/issues/source?bid=3244003&fileBranchId=3341467#l185)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java [hunk 6]
```diff
@@ -26,6 +26,7 @@
 import static br.ufc.quixada.npi.gpa.utils.Constants.REDIRECT_PAGINA_LISTAR_SELECAO;
 import static br.ufc.quixada.npi.gpa.utils.Constants.ABA_SELECIONADA;
 
+import java.awt.PageAttributes.MediaType;
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Codacy Issue found: [Avoid unused imports such as 'java.awt.PageAttributes.MediaType'](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2707229881/issues/source?bid=3244003&fileBranchId=3341467#l29)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java [hunk 7]
```diff
@@ -175,6 +178,22 @@ public String entrevista(@Valid @ModelAttribute("entrevista") Entrevista entrevi
 		return REDIRECT_PAGINA_LISTAR_SELECAO;
 	}
 	
+	
+	@RequestMapping(value = "consolidarTodos", method = RequestMethod.GET,  produces="application/json")
+	public @ResponseBody String consolidarTodos(@RequestParam("idSelecao") Integer idSelecao,@RequestParam("consolidacao") boolean consolidacao){
+		inscricaoService.consolidacaoDeTodos(idSelecao, consolidacao);
+		return string_json_vazio;
```
Comment (virginiafarias) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Não precisa retornar no formato de json, isso é feito automaticamente. Basta retornar uma string

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java [hunk 8]
```diff
@@ -40,7 +40,8 @@
 	
     @Inject
 	private GenericRepository<Inscricao> inscricaoRepository;
-
+    
+    private String string_consolidacao = "consolidacao";
```
Comment (virginiafarias) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java]: Não precisa criar essa string. Pode usar diretamente onde for necessário

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java [hunk 9]
```diff
@@ -175,6 +177,24 @@ public String entrevista(@Valid @ModelAttribute("entrevista") Entrevista entrevi
 		return REDIRECT_PAGINA_LISTAR_SELECAO;
 	}
 	
+	
+	@RequestMapping(value = "consolidarTodos", method = RequestMethod.GET,  produces=  MediaType.APPLICATION_JSON_VALUE)
+	public @ResponseBody Model consolidarTodos(@RequestParam("idSelecao") Integer idSelecao,@RequestParam("consolidacao") boolean consolidacao,Model model){
+		inscricaoService.consolidacaoDeTodos(idSelecao, consolidacao);
+		model.addAttribute("resultado","sucesso");
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Codacy Issue found: [The String "resultado" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2745786484/issues/source?bid=3244003&fileBranchId=3341467#l184)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java [hunk 10]
```diff
@@ -194,5 +192,26 @@ public void update(Integer idInscricao,boolean classificado) {
 		
 	}
 
+	@Override
+	@Transactional
+	public void consolidacaoDeTodos(Integer idSelecao, boolean consolidacao) {
+		Map<String, Object> params = new SimpleMap<String, Object>();
+		params.put("consolidacao", consolidacao);
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java]: Codacy Issue found: [The String "consolidacao" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2745786540/issues/source?bid=3244003&fileBranchId=3341467#l199)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/AlunoController.java [hunk 11]
```diff
@@ -21,6 +21,7 @@
 import static br.ufc.quixada.npi.gpa.utils.Constants.REDIRECT_PAGINA_MINHAS_INSCRICOES;
 
 import java.io.IOException;
+import java.io.UnsupportedEncodingException;
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/AlunoController.java]: Codacy Issue found: [Avoid unused imports such as 'java.io.UnsupportedEncodingException'](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2745786477/issues/source?bid=3244003&fileBranchId=3341467#l24)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java (hunk 9 above)
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Codacy Issue found: [The String "resultado" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2745869597/issues/source?bid=3244003&fileBranchId=3341467#l184)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java [hunk 12]
```diff
@@ -193,5 +191,26 @@ public void update(Integer idInscricao,boolean classificado) {
...
```
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java]: Codacy Issue found: [The String "consolidacao" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2745869656/issues/source?bid=3244003&fileBranchId=3341467#l198)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java (hunk 9 above)
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/controller/ServidorController.java]: Codacy Issue found: [The String "resultado" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2746207874/issues/source?bid=3244003&fileBranchId=3341467#l184)

File: gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java (hunk 12 above)
Comment (npiuser) [inline gpa-mae/src/main/java/br/ufc/quixada/npi/gpa/service/impl/InscricaoServiceImpl.java]: Codacy Issue found: [The String "consolidacao" appears 2 times in the file.](https://www.codacy.com/app/NPI-UFC/gpa-assuntos-estudantis/file/2746207933/issues/source?bid=3244003&fileBranchId=3341467#l198)

**Pull request merged**